packages/terra-table/src/Table.jsx

```diff
@@ -308,7 +310,25 @@ function Table(props) {
     return (hasSelectableRows ? [tableRowSelectionColumn] : []).concat(pinnedColumns).concat(overflowColumns);
   }, [hasSelectableRows, intl, onRowSelectionHeaderSelect, overflowColumns, pinnedColumns]);
 
-  const [tableColumns, setTableColumns] = useState(displayedColumns.map((column) => initializeColumn(column)));
+  // Create new displayedColumns object to pass to Section sub component to account for column spans
+  const displayedColumnsWithColumnSpan = [];
+  let i = 0;
+  displayedColumns.forEach((column) => {
+    if (column.columnSpan > 1) {
+      // Column highlighting is not supported for multiple column spans
+      displayedColumnsWithColumnSpan[i] = { ...column, columnSpanIndex: 0, columnHighlightColor: undefined};
+      i += 1;
+      for (let counter = column.columnSpan; counter > 1; counter -= 1) {
+        displayedColumnsWithColumnSpan[i] = { id: `${column.id}`, columnSpanIndex: (column.columnSpan - counter + 1) };
+        i += 1;
+      }
+    } else {
+      displayedColumnsWithColumnSpan[i] = column;
+      i += 1;
+    }
+  });
```

Comment:
Don't we have similar logic in the DataGrid components?  If we have the logic here why do we need it in those components.  Also, we should just be able to use a spread operator for the properties.  It seems like this can be simplified quite a bit.

Reply:
Removed this from Table and re-using object from DataGrid level